diff --git a/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb b/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb
index 7600829..94a8fb3 100644
--- a/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb
+++ b/node_modules/react-native/scripts/cocoapods/privacy_manifest_utils.rb
@@ -110,7 +110,7 @@ module PrivacyManifestUtils
                     accessed_api_types = content["NSPrivacyAccessedAPITypes"]
                     accessed_api_types.each do |accessed_api|
                     api_type = accessed_api["NSPrivacyAccessedAPIType"]
-                    reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+                    reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"] || []
                     used_apis[api_type] ||= []
                     used_apis[api_type] += reasons
                     end
